src/results.rs

```diff
@@ -71,3 +72,94 @@ pub(crate) struct GetMoreResult {
     pub(crate) batch: VecDeque<Document>,
     pub(crate) exhausted: bool,
 }
+
+/// Describes the type of data store returned when executing
+/// [`Database::list_collections`](../struct.Database.html#method.list_collections).
+#[derive(Debug, Clone, Serialize, PartialEq)]
+#[serde(rename_all = "lowercase", untagged)]
+#[non_exhaustive]
+pub enum CollectionType {
+    /// Indicates that the data store is a view.
+    View,
+
+    /// Indicates that the data store is a collection.
+    Collection,
+
+    /// An unknown collection type. This is included for forwards compatibility.
+    Other(String),
```

Comment:
given this command is part of the versioned API (which was part of our reasoning for why introducing such structs is pretty safe), is this something we actually need to worry about? 

the versioned API scope lists under "breaking changes":
> Add a new value to a StableCommand reply field's enum-like fixed set of values, e.g. a new index type (unless there's an opt-in mechanism besides API version).

a new collection type I would think falls into the same category as a new index type, BUT I will say I don't totally understand what an "opt-in mechanism" would be and if that negates our ability to ignore the "other" case here.

thoughts?

Reply:
that makes sense; sgtm